app/uk/gov/hmrc/transitmovementsvalidator/services/XmlValidationService.scala

```diff
@@ -76,38 +78,66 @@ class XmlValidationServiceImpl @Inject() (implicit ec: ExecutionContext) extends
             saxParser =>
               Using(source.runWith(StreamConverters.asInputStream(20.seconds))) {
                 xmlInput =>
-                  val parser = saxParser.newSAXParser.getParser
-
+                  val inputSource = new InputSource(xmlInput)
                   val errorBuffer: mutable.ListBuffer[XmlSchemaValidationError] =
                     new mutable.ListBuffer[XmlSchemaValidationError]
-
-                  parser.setErrorHandler(new ErrorHandler {
-                    override def warning(error: SAXParseException): Unit = {}
-
-                    override def error(error: SAXParseException): Unit =
-                      errorBuffer += XmlSchemaValidationError.fromSaxParseException(error)
-
-                    override def fatalError(error: SAXParseException): Unit =
-                      errorBuffer += XmlSchemaValidationError.fromSaxParseException(error)
-                  })
-
-                  val inputSource = new InputSource(xmlInput)
+                  var elementValue = ""
+                  var rootTag      = ""
+
+                  val parser = saxParser.newSAXParser.parse(
+                    inputSource,
+                    new DefaultHandler {
+                      var inMessageTypeElement = false
+
+                      override def characters(ch: Array[Char], start: Int, length: Int): Unit =
+                        if (inMessageTypeElement) {
+                          elementValue = new String(ch, start, length)
```

Comment:
What do I get if I put the following XML fragment?

```xml
<messageType>
    CC015C
</messageType>
```

specifically with the whitespace?

Reply:
if we spaces the validation works

Reply:
Awesome, thanks for checking!